tests/test_e2e.py

```diff
@@ -1,37 +1,66 @@
-import subprocess
 import pytest
+import subprocess
+import tempfile
+import yaml
 from kubernetes import client, config, utils
+from tests.utils import wait_for_deployments_ready
 
+E2E_INPUT_MANIFEST = "e2e-manifests/{}.yml"
+E2E_EXPECTED_YAML = "e2e-manifests/expected/{}.yml"
+E2E_RUNNER_IMAGE = "localhost:5000/illuminatio-runner:dev"
 
-@pytest.mark.e2e
-def test_deny_all_traffic_to_an_application():
-    namespace = "01-deny-all"
+
+@pytest.fixture
+def kubernetes_utils():
     config.load_kube_config()
     k8s_client = client.ApiClient()
-    corev1 = client.CoreV1Api()
+    return k8s_client, client.CoreV1Api(), client.AppsV1Api()
 
-    corev1.create_namespace(client.V1Namespace(
-        metadata=client.V1ObjectMeta(
-            name=namespace,
-            labels={"illuminatio-e2e": namespace})))
-    utils.create_from_yaml(k8s_client,
-                           "e2e-manifests/01-deny-all-traffic-to-an-application.yml",
-                           namespace=namespace)
-
-    # ToDo add sleep or wait until all resources are up otherwise we have a race condition
-    # ToDo handle execptions
-    res = subprocess.run(["illuminatio", "run", "--runner-image=localhost:5000/illuminatio-runner:dev"],
-                         capture_output=True,
-                         timeout=60)
 
-    assert res.returncode == 0
-    # ToDo evaluate result
-    print(res.stdout)
+@pytest.fixture(autouse=True)
+def clean_cluster(kubernetes_utils):
+    yield  # below code is executed after test(s)
+    _, core_v1, _ = kubernetes_utils
+    # delete e2e namespaces created in test setup
+    e2e_namespaces = core_v1.list_namespace(label_selector="illuminatio-e2e")
+    for namespace in e2e_namespaces.items:
+        core_v1.delete_namespace(name=namespace.metadata.name)
+    # delete illuminatio resources
+    subprocess.check_output(["illuminatio", "clean"])
 
-    # Clean up
-    res = subprocess.run(["illuminatio", "clean"], capture_output=True)
-    assert res.returncode == 0
-    print(res.stdout)
 
-    # Clean up
-    corev1.delete_namespace(name=namespace)
+@pytest.mark.parametrize(
+    "e2e_test_case",
+    [
+        "01-deny-all-traffic-to-an-application"
+        ],
+    )
+@pytest.mark.e2e
+def test__e2e__clean_setup__results_are_expected(e2e_test_case, kubernetes_utils):
+    # unpack kubernetes client
+    k8s_client, core_v1, apps_v1 = kubernetes_utils
```

Comment:
`core_v1` can be removed here since the `wait_for_deployments_ready` never uses the `core_api`